integration/flux_monitor_test

```diff
@@ -21,32 +28,29 @@ flux_monitor_test() {
   # Assert one job ran to update price
   assert "Flux Monitor Runs count" "chainlink -j runs list | jq length" $first_job_run_count
 
-  # TODO - once FM's check feed interval is configurable, update this test to change the
-  # feed's price and ensure FM initiates additional job run
-
   # nominally change price feed
-  # curl \
-  #   -H "Content-Type: application/json" \
-  #   --request PATCH \
-  #   --data '{"result":101}' \
-  #   "$EXTERNAL_ADAPTER_URL/result"
+  curl \
+    -H "Content-Type: application/json" \
+    --request PATCH \
+    --data '{"result":101}' \
+    "$EXTERNAL_ADAPTER_URL/result"
 
-  # sleep 20
+  sleep 5
 
   # Assert no jobs ran after nominal update
-  # assert "Flux Monitor Runs count" "chainlink -j runs list | jq length" $first_job_run_count
+  assert "Flux Monitor Runs count" "chainlink -j runs list | jq length" $first_job_run_count
```

Comment:
Might be better here to have something loop and assert for 5 seconds, rather than sleeping. I believe we already have that logic in the bash code.

Reply:
Looks like you are right B)

Reply:
The assert function defined in `integration/common` returns as soon as it finds a match, and does not immediately fail if the expression does not evaluate to the expectation value. In this scenario, I think we want the inverse functionality. We want to keep checking continuously over at least a 5 second period to make sure the expression always evaluates to the expectation value. I might suggest adding another function to `common` like this:

```bash
assertFor () {
  sleepCount=0
  while [ "$sleepCount" -le "$4" ] && output=`eval $2`; do
    if [ "$output" != "$3" ]; then
      printf -- "\033[31mTEST FAILED!\033[0m $1: expected $3, actual $output; while evaluating '$2'.\n"
      exit 1
      return
    fi
    sleepCount=$((sleepCount+1))
    sleep 1
  done
  printf -- "\033[32mTest passed!\033[0m $1: got expected value $3 over $4 second period.\n"
  return
}
```

and then here we can use it like so

```bash
# wait polling interval + 2s to allow time for FM to make wait for polling interval, post tx,
# and for tx to get confirmed
assertFor "Flux Monitor Runs count" "chainlink -j runs list | jq length" $first_job_run_count 7
```

Reply:
@RyanRHall thanks for spotting this.  Looks like we both overlooked it.  In this case wouldn't it be simpler to sleep beforehand?  Its essentially the same thing, but without bailing out on a failing case which should be very rare.

Its odd testing for non-behavior

Reply:
Yeah agreed it's the same as sleeping, and I'm fine with just adding a sleep :+1::sleeping: .

Reply:
It's different to sleeping in that it can't return early if the test happens faster. If you change `sleep 1` to `sleep 0.2` in your example bash code above, and use that, it can make the test happen as fast as possible. Which is an important step in keeping our overall test suite fast!

Reply:
@j16r the situation we're referring to requires a bit of a different approach. We want to make sure that a value does not change, even after X seconds. So we have to wait the full X seconds and then check. If we use assert, the value might be fine at first, but then change later and we need to fail in that situation.

Reply:
Ahh, this is a negative assertion, I missed that - carry on then.